jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/inputs/ButtonTests.java

```diff
@@ -27,138 +27,77 @@ public void before() {
         buttonPage.isOpened();
     }
 
-    @Test
-    public void defaultButtonTest() {
-
-        containedButtons.get(1).click();
-        containedButtons.get(1).is().text("DEFAULT");
-        containedButtons.get(6).is().text("Last click: Default");
-
-        containedButtons.get(2).click();
-        containedButtons.get(2).is().text("PRIMARY");
-        containedButtons.get(6).is().text("Last click: Primary");
-
-        containedButtons.get(3).click();
-        containedButtons.get(3).is().text("SECONDARY");
-        containedButtons.get(6).is().text("Last click: Secondary");
-
-        containedButtons.get(4).is().text("DISABLED");
-        containedButtons.get(4).is().attr("disabled");
-        containedButtons.get(4).is().disabled();
-
-        containedButtons.get(5).click();
-        containedButtons.get(5).is().text("LINK");
-        containedButtons.get(6).is().text("Last click: Link");
-        containedButtons.get(5).is().notVisible();
+    @Test(dataProvider = "containedButtonTestData", dataProviderClass = ButtonDataProvider.class)
+    public void containedButtonTest(int buttonIndex, String buttonTitle, boolean isEnable){
+        MUIButton containedButton = containedButtons.get(buttonIndex);
+        containedButton.label().is().text(buttonTitle);
+        if(isEnable) {
+            containedButton.is().enabled();
+            containedButton.click();
+        } else {
+            containedButton.is().disabled();
+        }
     }
 
-    @Test
-    public void textButtonTest() {
-
-        textButtons.get(1).click();
-        textButtons.get(1).is().text("DEFAULT");
-        textButtons.get(6).is().text("Last click: Default");
-
-        textButtons.get(2).click();
-        textButtons.get(2).is().text("PRIMARY");
-        textButtons.get(6).is().text("Last click: Primary");
-
-        textButtons.get(3).click();
-        textButtons.get(3).is().text("SECONDARY");
-        textButtons.get(6).is().text("Last click: Secondary");
-
-        textButtons.get(4).is().text("DISABLED");
-        textButtons.get(4).is().attr("disabled");
-        textButtons.get(4).is().disabled();
-
-        textButtons.get(5).click();
-        textButtons.get(5).is().text("LINK");
-        textButtons.get(6).is().text("Last click: Link");
-        textButtons.get(5).is().notVisible();
+    @Test (dataProvider = "textButtonTestData", dataProviderClass = ButtonDataProvider.class)
+    public void textButtonTest(int buttonIndex, String buttonTitle, boolean isEnable) {
+        MUIButton textButton = textButtons.get(buttonIndex);
+        textButton.label().is().text(buttonTitle);
+        if(isEnable) {
+            textButton.is().enabled();
+            textButton.click();
+        } else {
+            textButton.is().disabled();
```

Comment:
As opposed to an enabled button, maybe it's worth checking that nothing happens when that button is clicked?

Reply:
Attempting click a disabled button will make test fail.